frontend/common/columns.tsx

```diff
@@ -185,6 +187,48 @@ export function useOrganizationNameColumn(options?: {
   );
   return column;
 }
+export function useStatusColumn(options?: {
+  tooltip?: string;
+  disableLinks?: boolean;
+  disableSort?: boolean;
+}) {
+  const { t } = useTranslation();
+  const column: ITableColumn<{
+    type?: string;
+    status?: string;
+    summary_fields?: {
+      last_job?: {
+        id?: number;
+      };
+      current_job?: {
+        id?: number;
+      };
+    };
+  }> = useMemo(
+    () => ({
+      header: t('Status'),
+      cell: (item) => (
+        <Tooltip content={options?.tooltip ?? ''} position="top">
+          <StatusCell
+            status={item.status}
+            to={RouteObj.JobOutput.replace(':job_type', item.type ? item.type : '').replace(
+              ':id',
+              (
+                item.summary_fields?.current_job?.id ??
+                item.summary_fields?.last_job?.id ??
+                ''
+              ).toString()
+            )}
+            disableLinks={options?.disableLinks}
```

Comment:
This enhancement looks great and resolves the vertical row alignment issues we had with status and project type table cells. 

I have two findings:
1) We should disable the link to the project's sync job when it is missing `summary_fields.current_job` and `summary_fields.last_job`. Currently a project without a job update shows a page not found. 
2) It would be nice if we could make the useStatusColumn hook more generic so that we could reuse it in Inventories. If we include arguments for the link and status, then I think this could also be plugged into the inventories list. I don't think this is necessary for this PR to merge, just a suggestion.